activestorage/lib/active_storage/attached/one.rb

```diff
@@ -10,11 +10,11 @@ class Attached::One < Attached
     # You don't have to call this method to access the attachment's methods as
     # they are all available at the model level.
     def attachment
-      record.public_send("#{name}_attachment")
+      change.present? ? change.attachment : record.public_send("#{name}_attachment")
```

Comment:
`change&.attachment || record.public_send("#{name}_attachment")`?

Reply:
Tricky: we don’t want to fall back to the attachment in the DB when a `Delete*` change is pending.